### ISSUE TITLE
valid-jsdoc false positive for class documentation with class property arrow function

### ISSUE THREAD

**Issue by amannn** [babel/babel-eslint#518]
I accidentally [raised an issue over at the eslint repo](https://github.com/eslint/eslint/issues/9308), but apparently it belongs better in this project. Here's the report:

**Tell us about your environment**

* **ESLint Version:** 4.6.1
* **Node Version:** 7.10.1
* **npm Version:** 4.2.0 (yarn 1.0.1)

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

```js
{
  "parser": "babel-eslint",
  "rules": {
    "valid-jsdoc": 2
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
/**
 * Class description
 */
class Component {
  onClick = e => console.log(e);
}
```

**What did you expect to happen?**
No lint errors.

**What actually happened? Please include the actual, raw output from ESLint.**
At line number 1 the following errors are triggered:

 - `Missing JSDoc @returns for function. (valid-jsdoc)`
 - `Missing JSDoc for parameter 'e'. (valid-jsdoc)`

It looks like eslint thinks the comment above the class refers to the arrow function. If I replace the arrow function with a regular class method, the error vanishes. The error is also there when there's other methods above the class property function.

**Comment by markwoon** [comment]
I'm seeing the exact same thing.  Eslint's default parser handles this properly.

**Comment by DustinReagan** [comment]
I'm having the same problem.

**Comment by philostler** [comment]
Also experiencing this ATM while moving to class property arrow functions.

**Comment by philostler** [comment]
There are a handful of issues involving `valid-jsdoc` sitting idle for year(s). What are the blockers to getting these resolved?

**Comment by kaicataldo** [comment]
Just so everyone’s aware, ESLint core has decided to [deprecate the valid-jsdoc rule](https://eslint.org/blog/2018/11/jsdoc-end-of-life).

The current logic for locating JSDoc comments is actually built into core, which made it impossible for parsers like babel-eslint to provide a way to find these rules. Check out the plugin suggested in the blog post and, if it doesn’t currently support this case, consider making an issue for that project.

**Comment by kaicataldo** [comment]
Thank you for the issue. Now that [`@babel/eslint-parser` has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).
